### PR TITLE
feat(listbox): add new component

### PR DESCRIPTION
**Pull request by rinaok**
after https://github.com/Vonage/vivid-3/pull/760 will be merged

### CODE REVIEW

**Review by yinonov — CHANGES_REQUESTED** [review]
@rinaok there are 2 issues with focus needs fixing
- [ ] clicking on listbox items doesn't activate (focus) listbox
- [ ] when extending listboxElement, listbox element isn't activated (focused). extending listbox enabling it to be activated

**Comment by yinonov** [comment]
listbox options, when are selected / checked, are styled by background color only. does it comply with rules? should we add another indicator?

image

File: libs/components/src/lib/listbox-option/listbox-option.scss [hunk 1]
```diff
@@ -3,14 +3,14 @@
...
-	$shades: primary faint soft,
+	$shades: primary primary-increment faint soft,
 	$default: accent
 );
 @use "../../../../shared/src/lib/sass/mixins/connotation" as connotation;
 
 @use "../../../../shared/src/lib/sass/mixins/appearance/config" as appearance-config with (
 	$appearances: ghost,
-	$states: idle hover disabled active selected,
+	$states: idle hover disabled active selected selectedAndHover,
```
Comment (rinaok) [inline libs/components/src/lib/listbox-option/listbox-option.scss:13]: There is a new PR for option, it's better to make the changes there. 
https://github.com/Vonage/vivid-3/pull/760

File: libs/components/src/lib/listbox-option/listbox-option.scss [hunk 2]
```diff
@@ -23,6 +23,7 @@
 	@include connotation.connotation;
 	@include appearance.appearance;
 
+	position: relative;
```
Comment (rinaok) [inline libs/components/src/lib/listbox-option/listbox-option.scss:26]: @rachelbt @yinonov This had to be returned. Without it, the `Multiple selection` is not working.

File: libs/components/src/lib/listbox/listbox.scss [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+@use '../../../../shared/src/lib/sass/mixins/appearance/config' as appearance-config with (
+	$appearances: fieldset ghost,
+	$states: idle disabled,
+	$default: fieldset,
+);
+@use '../../../../shared/src/lib/sass/mixins/appearance' as appearance;
+
+:host {
+	display: inline-flex;
```
Comment (rachelbt) [inline libs/components/src/lib/listbox/listbox.scss:17]:
```suggestion

```
I don't think this is needed, and this prevent the list-box getting its parent width

File: libs/components/src/lib/listbox-option/listbox-option.scss (hunk 2 above)
Comment (yinonov) [reply libs/components/src/lib/listbox-option/listbox-option.scss:26]: and the focus indicator needs it as well

File: libs/components/src/lib/listbox/listbox.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	@attr appearance?: LisboxAppearance;
+
+	override slottedOptionsChanged(prev: Element[] | undefined, next: Element[]) {
+		super.slottedOptionsChanged(prev, next);
+		this.#disableSlottedChildren();
+	}
+
+	override attributeChangedCallback(name: string, oldValue: string, newValue: string): void {
+		super.attributeChangedCallback(name, oldValue, newValue);
+		switch (name) {
```
Comment (olaf-k) [inline libs/components/src/lib/listbox/listbox.ts:35]: a simple `if` should be enough for the time being

**Review by olaf-k — CHANGES_REQUESTED** [review]

**Review by olaf-k — DISMISSED** [review]
👍

**Review by yinonov — CHANGES_REQUESTED** [review]
wait for #750

**Comment by rinaok** [comment]
> wait for #750

merged

**Review by yinonov — CHANGES_REQUESTED** [review]
mouse click on an option should set focus on listbox. additional keyboard events like arrow navigation should continue navigating through other options but it doesn't

**Comment by rinaok** [comment]
> mouse click on an option should set focus on listbox. additional keyboard events like arrow navigation should continue navigating through other options but it doesn't

@yinonov @rachelbt If I return [this](https://github.com/Vonage/vivid-3/pull/635/commits/1c0ca1ffcdfda0e5164c722584d57ce957082eda) , it fixes the problem.

**Review by rachelbt — APPROVED** [review]
LGTM

**Comment by rinaok** [comment]
@yinonov The [refactor](https://github.com/Vonage/vivid-3/commit/2e55fe2331d0d07bd0a88650a3cbf4586a308499#diff-590cb97e4da80dd05a7a1d55cd2d0a3ab33e35cdc20e565a834b949c622a63e7) in fieldset broke the listbox. Can you take a look?

**Review by yinonov — APPROVED** [review]

**Pull request merged**